invenio_banners/resources/resource.py

```diff
@@ -30,41 +36,82 @@ def create_url_rules(self):
         routes = self.config.routes
         return [
             route("POST", routes["create"], self.create),
+            route("GET", routes["banner"], self.read),
+            route("GET", routes["list"], self.search),
+            route("DELETE", routes["banner"], self.delete),
+            route("PUT", routes["banner"], self.update),
         ]
 
-    #
-    # Primary Interface
-    #
+    @request_view_args
+    @request_data
+    @response_handler()
+    def update(self):
+        """Update a banner."""
+        banner_id = resource_requestctx.view_args["banner_id"]
+        banner = self.service.update(
+            id=banner_id,
+            identity=g.identity,
+            data=resource_requestctx.data,
+        )
+
+        return banner.to_dict(), 200
+
     @request_view_args
     @response_handler()
     def read(self):
-        """Read an item."""
-        # TODO: not implemented/tested yet
+        """Read a banner."""
+        banner_id = resource_requestctx.view_args["banner_id"]
+        banner = self.service.read(
+            id=banner_id,
+            identity=g.identity,
+        )
 
-        item = self.service.read(url_path=resource_requestctx.view_args["url_path"])
-        return item.to_dict(), 200
+        return banner.to_dict(), 200
 
     @request_search_args
     @request_view_args
+    @request_extra_args
     @response_handler(many=True)
     def search(self):
-        """Perform a search over banners."""
-        # TODO: not implemented/tested yet
+        """Perform a search over the banners."""
+        active_arg = resource_requestctx.args.get("active")
 
-        banners = self.service.search(
-            identity=g.identity,
-            params=resource_requestctx.args,
-        )
+        if active_arg is None:
```

Comment:
given that it is a `Boolean`, it should only have values `[True|False]`, right?
I would change to:
```suggestion
        if active_arg:
            ...
        else:
            ...
```